src/contexts/notes-context/notes-reducer.jsx

```diff
@@ -31,17 +41,109 @@ export default function notesReducer(state, { type, payload }) {
       return {
         ...state,
         notes: payload,
-        note_editor: {},
+        note_editor: {
+          title: "",
+          body: "",
+          priority: "medium",
+          labels: [],
+          createdAt: formatDate(),
+          cardColor: "white",
+        },
+      };
+    }
+
+    case "MOVE_TO_TRASH": {
+      return {
+        ...state,
+        notes: payload.notes,
+        note_editor: {
+          title: "",
+          body: "",
+          priority: "medium",
+          labels: [],
+          createdAt: formatDate(),
+          cardColor: "white",
+        },
+        trash: [...state.trash, payload.note],
+      };
+    }
+
+    case "DELETE_NOTE": {
+      return {
+        ...state,
+        notes: payload.notes,
+        trash: state.trash.filter(item => item._id !== payload.delete_id),
+      };
+    }
+
+    case "RESTORE_NOTE": {
+      return {
+        ...state,
+        notes: payload.notes,
+        trash: state.trash.filter(item => item._id !== payload.restore_id),
+      };
+    }
+    case "GET_ARCHIVED_NOTES": {
+      return {
+        ...state,
+        archives: payload,
       };
     }
 
-    case "UPDATE_NOTES": {
-      state.trash.push(state.notes.find(item => item._id === payload.note_id));
+    case "MOVE_TO_ARCHIVE": {
       return {
         ...state,
         notes: payload.notes,
-        note_editor: {},
+        archives: payload.archives,
       };
     }
+
+    case "DELETE_FROM_ARCHIVE": {
+      return {
+        ...state,
+        trash: [
+          ...state.trash,
+          state.archives.find(item => item._id === payload.note_id),
+        ],
+        archives: payload.archives,
+      };
+    }
+
+    case "ADD_NEW_TAG": {
+      return {
+        ...state,
+        tags: [...state.tags, payload],
```

Comment:
You can have a check on the payload value whether it is already present in the tag List then no need to update the tags and if it is new value you can update the tag List. This way you can avoid adding same tag twice!!